ttg/ttg/parsec/device.h

```diff
@@ -35,6 +36,12 @@ namespace ttg_parsec {
     }
   } // namespace detail
 
+
+  inline
+  int num_devices() {
+    return parsec_nb_devices - detail::first_device_id;
```

Comment:
We now have `parsec_context_query(ctx, PARSEC_CONTEXT_QUERY_DEVICES, PARSEC_DEV_CUDA)` that can be used to count devices without having to know about the first device id.

Reply:
I still need the first device ID for the conversion between TTG device IDs (matching the CUDA/HIP/L0 ID space) and parsec's ID space. What is the best way for doing that without knowing the first device ID?

Reply:
I don't think we have something nice for that atm. 

BTW you may still want to cache the result because context_query is running a loop.